### ISSUE TITLE
Zoho - Invalid Client

### ISSUE THREAD

**Issue by c4tom** [rclone/rclone#5029]
First of all, thanks for adding this zoho option.

I tried to configure a zoho remote drive, with this new version 1.5.4 and I can test 2 types of zoho licenses, Zoho One and Zoho Workplace, and both, I got the message

```
Invalid Client

The transmitted client ID does not exist
```

> Storage> 39 
> ** See help for zoho backend at: https://rclone.org/zoho/ **
> 
> Zoho region to connect to. You'll have to use the region you organization is registered in.
> Enter a string value. Press Enter for the default ("").
> Choose a number from below, or type in your own value
>  1 / United states / Global
>    \ "com"
>  2 / Europe
>    \ "eu"
>  3 / India
>    \ "in"
>  4 / Australia
>    \ "com.au"
> region> 1
> Edit advanced config? (y/n)
> y) Yes
> n) No (default)
> y/n> 
> Remote config
> Use auto config?
>  * Say Y if not sure
>  * Say N if you are working on a remote or headless machine
> y) Yes (default)
> n) No
> y/n> 
> If your browser doesn't open automatically go to the following link: http://127.0.0.1:53682/auth?state=wF0U_y_ZIGLZycBnv2zqxA
> Log in and authorize rclone for access
> Waiting for code...

**Comment by ncw** [comment]
I wonder if the client_id's are regional? We've been testing with the EU region.

What do you think @buengese

**Comment by c4tom** [comment]
> I wonder if the client_id's are regional? We've been testing with the EU region.
> 
> What do you think @buengese

i tried with EU, but, not work

https://accounts.zoho.com/oauth/v2/auth?access_type=offline&client_id=**1000.OZNFWW075EKDSIE1R42HI9I2SUPC9A**&prompt=consent&redirect_uri=http%3A%2F%2Flocalhost%3A53682%2F&response_type=code&scope=aaaserver.profile.read+WorkDrive.team.READ+WorkDrive.workspace.READ+WorkDrive.files.ALL&state=FqjKOuka0obkE8tS3h2KKg

Note: is client_id=**1000.OZNFWW075EKDSIE1R42HI9I2SUPC9A** generated?


Note 2: I followed the example of the page https://rclone.org/zoho/, but there is no place in this version, which asks

client_id>
client_secret>

**Comment by buengese** [comment]
Sorry for being so slow to respond. The client's are indeed regional though there should be an option to create one client id for all regions. I'm not entirely sure if that's still my clientID that is currently being used. Client id and secret not being configurable is indeed a problem because I would recommend creating you own client id for Zoho I'll probably fix this tomorrow.

**Comment by buengese** [comment]
I've fixed setting custom client ID's which will be merged soon. I've also confirmed that the current client id does in fact work for the eu region. In which region is your account registered?

**Comment by c4tom** [comment]
thank you, i use US region

**Comment by ivandeex** [comment]
The fix was merged yesterday by @buengese 
@c4tom Please try latest beta zip for your architecture from https://beta.rclone.org

**Comment by c4tom** [comment]
sorry, the delay, i will test

**Comment by buengese** [comment]
Not sure why this is happening. It's definitely working for me. I've created another build with a different clientid you can download it [here](https://beta.rclone.org/branch/zoho-clientid-test/v1.55.0-beta.5250.61316f4ef.zoho-clientid-test/) and test if that works. You can also try setting up your own client id at [https://api-console.zoho.com/](https://api-console.zoho.com/)

**Comment by c4tom** [comment]
@buengese 

I was able to configure it with this version that you indicated, however, it does not list and no folder or file appears

I chose the US region

Note: I did not need to create any new configuration in https://api-console.zoho.com/

**Comment by buengese** [comment]
I finally created a test US account which I really should've done a long time ago. I replaced the API key with a new one which definetly works in all regions. This sould've fixed the problem but addionally the region selectoin was also broken which caused the eu api endpoints to be used for all regions. I'm not sure how I missed that but it's fixed now. That means the [latest beta ](https://beta.rclone.org/v1.55.0-beta.5320.da5b0cb61/) should really, definetly, for sure work now. Sorry that took me so long.
